pandas/tests/arrays/test_datetimelike.py

```diff
@@ -409,3 +444,16 @@ def test_int_properties(self, period_index, propname):
         expected = np.array(getattr(pi, propname))
 
         tm.assert_numpy_array_equal(result, expected)
+
+    def test_array(self, period_index):
+        arr = PeriodArray(period_index)
+
+        result = np.asarray(arr)
```

Comment:
what about other dtypes?

Reply:
object and int are tested below, which others do you want?

Reply:
well, ``str``, ``float``, basically ones which are weird

Reply:
Though, if wed don't explicitly handle them in our `__array__` NumPy takes care of it.